service/worker/scheduler/fx.go

```diff
@@ -87,15 +91,19 @@ func (s *workerComponent) DedicatedWorkerOptions(ns *namespace.Namespace) *worke
 	}
 }
 
-func (s *workerComponent) Register(worker sdkworker.Worker, ns *namespace.Namespace) {
+func (s *workerComponent) Register(worker sdkworker.Worker, ns *namespace.Namespace, details workercommon.RegistrationDetails) {
 	worker.RegisterWorkflowWithOptions(SchedulerWorkflow, workflow.RegisterOptions{Name: WorkflowType})
-	worker.RegisterActivity(s.activities(ns.Name(), ns.ID()))
+	worker.RegisterActivity(s.activities(ns.Name(), ns.ID(), details.WorkerFactor))
 }
 
-func (s *workerComponent) activities(name namespace.Name, id namespace.ID) *activities {
+func (s *workerComponent) activities(name namespace.Name, id namespace.ID, workerFactor float64) *activities {
+	localRPS := func() float64 {
+		return s.globalNSStartWorkflowRPS(name.String()) * workerFactor
```

Comment:
Can we not introduce this workerFactor? I think it might be more clear here to calculate it as:
localRPS = multiplicity * (globalRPS / totalWorkerCount).

Reply:
So you want to pass in multiplicity and total and do the division inside here? I guess that's fine too